pkg/mflag/flag.go

```diff
@@ -557,21 +578,30 @@ func (fs *FlagSet) PrintDefaults() {
 				names = append(names, name)
 			}
 		}
+
 		if len(names) > 0 && len(flag.Usage) > 0 {
-			val := flag.DefValue
 
-			if home != "" && strings.HasPrefix(val, home) {
-				val = homedir.GetShortcutString() + val[len(home):]
+			nstring := fmt.Sprintf("-%s", strings.Join(names, ", -"))
+			if len(names) == 1 && strings.HasPrefix(names[0], "-") {
+				nstring = "    " + nstring
 			}
 
-			if isZeroValue(val) {
-				format := "  -%s"
-				fmt.Fprintf(writer, format, strings.Join(names, ", -"))
+			if isZeroValue(flag.DefValue) && flag.Value.Type() == "BOOL" {
+				fmt.Fprintf(writer, "  %s", nstring)
 			} else {
-				format := "  -%s=%s"
-				fmt.Fprintf(writer, format, strings.Join(names, ", -"), val)
+				fmt.Fprintf(writer, "  %s=%s", nstring, flag.Value.Type())
+			}
+
+			usage := flag.Usage
+			if !isZeroValue(flag.DefValue) {
+				val := flag.DefValue
+				if home != "" && strings.HasPrefix(val, home) {
+					val = homedir.GetShortcutString() + val[len(home):]
+				}
+
+				usage = fmt.Sprintf("%s, default is '%s'", usage, val)
```

Comment:
Unfortunately these format changes will make it highly unlikely the help output will ever get back down to 80 chars.  The type info, double-dash alignment padding and more verbose default value output adds 15+ chars to several options; in a few cases the changes result in an increase of ~25 chars.  I tweaked several option descriptions to get down to 103 chars, but shedding another 20+ chars isn't feasible without making several descriptions more or less useless.

Assuming only experienced Docker users will benefit from extremely terse output, what about adding an environment variable to allow the user to suppress type, usage and/or default information in the help output, e.g. [https://gist.github.com/sean-jc/1f37ca226097212f3233362151301fe0](url)?  This would allow some amount of customization without delving into templates or other heavy approaches.

For example, if you are familiar with the options but can never remember the exact syntax or default values:
```
Usage: docker [OPTIONS] COMMAND [arg...]
       docker [ --help | -v | --version ]

A self-sufficient runtime for containers.

Options:

      --config=STRING       Default is '~/.docker'
  -D, --debug               
  -H, --host=LIST           
  -h, --help                
  -l, --log-level=STRING    Default is 'info'
      --tls                 
      --tlscacert=STRING    Default is '~/.docker/ca.pem'
      --tlscert=STRING      Default is '~/.docker/cert.pem'
      --tlskey=STRING       Default is '~/.docker/key.pem'
      --tlsverify           
  -v, --version             
```